src/main/java/dinkplugin/notifiers/KillCountNotifier.java

```diff
@@ -33,27 +41,47 @@ protected String getWebhookUrl() {
         return config.killCountWebhook();
     }
 
+    public void reset() {
+        this.data = null;
+    }
+
     public void onGameMessage(String message) {
         if (isEnabled())
-            parse(message).ifPresent(pair -> handleKill(pair.getLeft(), pair.getRight(), message));
+            parse(message).ifPresent(this::updateData);
+    }
+
+    public void onFriendsChatNotification(String message) {
+        // For CoX, Jagex sends duration via FRIENDSCHATNOTIFICATION
+        if (message.startsWith("Congratulations - your raid is complete!"))
+            this.onGameMessage(message);
     }
 
-    private void handleKill(String boss, int killCount, String rawMessage) {
-        if (!checkKillInterval(killCount)) return;
+    public void onTick() {
+        if (data != null) {
+            // all data must be sent on the same tick to be included
+            handleKill(data);
+            reset();
+        }
+    }
+
+    private void handleKill(BossNotificationData data) {
+        if (data.getBoss() == null || data.getCount() == null || !checkKillInterval(data.getCount(), data.isPersonalBest()))
+            return;
```

Comment:
Can we split this up into multiple if-cases, probably 2
1 to check if all required data is present, 1 to check if we should send the message based on the kill interval or pb state